commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -603,3 +603,36 @@ def retire_customer_anonymize_fields(self, customer_id: str, customer_version: i
                          f"with ID: {customer_id}, after LMS retirement with "
                          f"error correlation id {err.correlation_id} and error/s: {err.errors}")
             raise err
+
+    def is_first_time_discount_eligible(self, email: str) -> bool:
+        """
+        Check if a user is eligible for a first time discount
+        Args:
+            email (str): Email of the user
+        Returns (bool): True if the user is eligible for a first time discount
+        """
+        try:
+            discounts = self.base_client.discount_codes.query(
+                where="code in :discountCodes",
+                predicate_var={'discountCodes': settings.COMMERCETOOLS_FIRST_TIME_DISCOUNTS}
+            )
+            discount_ids = [discount.id for discount in discounts.results]
+
+            discounted_orders = self.base_client.orders.query(
+                where=[
+                    "customerEmail=:email",
+                    "orderState=:orderState",
+                    "discountCodes(discountCode(id in :discountIds))"
+                ],
+                predicate_var={'email': email, 'discountIds': discount_ids, 'orderState': 'Complete'}
```

Comment:
Just making sure that we don't want to cater for refunded orders here right?

Reply:
Refunded orders are not handled by CT itself, and if I remember correctly, we had to check for their first usage only. Correct me if I am wrong